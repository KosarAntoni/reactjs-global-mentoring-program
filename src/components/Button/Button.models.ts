export interface ButtonProps {
  children: string
  style?: 'transparent' | 'solid' | 'outline'
  className?: string
}
