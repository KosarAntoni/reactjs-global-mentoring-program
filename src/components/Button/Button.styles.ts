import styled from '@emotion/styled'

export const StyledButton = styled.button`
  cursor: pointer;

  padding: 1.8rem 7.5rem;

  color: var(--white);
  text-transform: uppercase;

  background: var(--pastele-red);
  border: none;
  border-radius: 4px;

  &.outline {
    color: var(--pastele-red);

    background: transparent;
    border: 1px solid var(--pastele-red);
  }

  &.transparent {
    padding: 1.1rem 1.8rem;

    color: var(--pastele-red);

    background: rgb(96 96 96 / 68%);
  }
`
