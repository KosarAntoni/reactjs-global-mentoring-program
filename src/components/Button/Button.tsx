import React, { FC } from 'react'
import classNames from 'classnames'

import { StyledButton } from './Button.styles'

import { ButtonProps } from './Button.models'

const Button: FC<ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>> =
  ({ children, style = 'solid', className, ...props }) => (
    <StyledButton
      className={classNames(style, className)}
      {...props}
    >
      {children}
    </StyledButton>
  )

export default Button
