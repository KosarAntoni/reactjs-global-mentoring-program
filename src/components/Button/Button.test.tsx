import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'

import Button from './Button'

import '@testing-library/jest-dom'

const testFunction = jest.fn()

test('the Button component renders correctly with required and optional props', () => {
  render(<Button className='test-class' onClick={testFunction}>Test</Button>)

  const buttonNode = screen.getByRole('button', { name: 'Test' })

  // the component has been rendered
  expect(buttonNode).toBeInTheDocument()

  // generating a snapshot, not for testing purposes but only to see what changes in the HTML structure during CR if the developer updates the snapshot
  expect(buttonNode).toMatchSnapshot()

  // after click right function is called
  fireEvent.click(buttonNode)
  expect(testFunction).toBeCalled()

  // the comonent has correct classes
  expect(buttonNode).toHaveClass('button button--solid test-class')
})
