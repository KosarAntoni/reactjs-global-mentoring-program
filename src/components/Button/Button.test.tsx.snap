// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Button component renders correctly with required and optional props 1`] = `
<button
  class="button button--solid test-class"
>
  Test
</button>
`;
